Real Use Case for Automating QFieldCloud Project Management

SeqLaz opened the pull request.

suricactus reviewed (commented):
Few general comments:

- remove the numbering, you can say: "here it is typical userstory".
- one sentence per line
- you can skip the detailed explanation on cronjob and redirect to https://cron.guru
- add two tabs that say "Linux (bash)" and one for "Windows".
- Incluce the output of the command in a follow-up codeblock.

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+
+### **Other Useful QFieldCloud CLI Commands**
+
+#### **List Your Projects**
+
+To list all projects associated with your account:
+
+```shell
+qfieldcloud-cli list-projects
+```
+
+To include public projects in the list:
+
+```shell
+qfieldcloud-cli list-projects --include-public
+```
````
Move this up after the "Create a project" section. With a text like:

> If you forgot to copy your project ID, you can always check the list of all the projects on QFieldCloud.

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+    $env:QFIELDCLOUD_TOKEN = "123abcXYZ987exampleToken"
+    ```
+
+#### Create a project
````
```suggestion
### Create a project
```

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+qfieldcloud-cli list-projects --include-public
+```
+
+#### **List Files in a Project**
+
+To view all files in a specific project:
+
+```shell
+qfieldcloud-cli list-files "123e4567-e89b-12d3-a456-426614174000"
+```
````
This should go after "Upload Local Files to QFieldCloud".

> Now you can see your files on QFieldCloud...

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+qfieldcloud-cli delete-project "123e4567-e89b-12d3-a456-426614174000"
+```
+
+#### **Manage Collaborators**
+
+You can add, remove, or change the roles of collaborators on your project.
+
+- **Add a Collaborator:**
+
+```shell
+qfieldcloud-cli collaborators-add "123e4567-e89b-12d3-a456-426614174000" "ninja007" admin
+```
+
+- **Remove a Collaborator:**
+
+```shell
+qfieldcloud-cli collaborators-remove "123e4567-e89b-12d3-a456-426614174000" "ninja007"
+```
+
+- **Change a Collaborator’s Role:**
+
+```shell
+qfieldcloud-cli collaborators-patch "123e4567-e89b-12d3-a456-426614174000" "ninja001" editor
+```
````
Move this before "Schedule and Trigger a Package Job".

> QFieldCloud is a collaborative platform, so you can manage your fieldworks....

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+qfieldcloud-cli list-files "123e4567-e89b-12d3-a456-426614174000"
+```
+
+#### **Delete a Project**
+
+To permanently delete a project (be cautious—this action cannot be undone):
+
+```shell
+qfieldcloud-cli delete-project "123e4567-e89b-12d3-a456-426614174000"
+```
````
You can keep this a last section.

> Once you are done with a project, you can delete it...

suricactus reviewed (changes requested):
Ok, almost there. A few comments:

- Rename "Linux" to "Bash" and "Windows" to "PowerShell"
- Add both version for all commands.
- Also put somewhere an advanced example using `qfieldcloud-cli --json login | jq .session_token` or something like this.

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+    47 8 * * * qfieldcloud-cli job-trigger '123e4567-e89b-12d3-a456-426614174000' package
+    ```
+
+    This triggers the package job daily at the specified time. For more information about [cronjob](https://crontab.guru/)
````
```suggestion
    This triggers the package job daily at the specified time. For more information about [cronjob](https://crontab.guru/).
```

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+    schtasks /create /tn "QFieldCloud Job Trigger" /tr "qfieldcloud-cli job-trigger '123e4567-e89b-12d3-a456-426614174000' package" /sc daily /st 08:47
+    ```
+
+    This triggers the package job daily at the specified time. For more information about [schtasks](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/schtasks)
````
```suggestion
    This triggers the package job daily at the specified time. For more information about [schtasks](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/schtasks).
```

suricactus commented on docs/docs/examples.md:
```diff
@@ -0,0 +1,265 @@
+
+### **Monitor Job Status**
+
+After triggering a job, monitor its status to ensure successful completion:
+
```
> first you can get a list of all the jobs for the specific projects

> `list-jobs`

> then you can check the status of a specific job

> `job-status`

> you can continiously check for job status to be finished like this:

```
while $JOB_STATUS not  in ["success", "failed"]; do
  JOB_STATUS=$(qfieldcloud-cli --json job-status 8630ebc2-b6e3-41f9-a370-2b0838d8daa4 | jq '.output' -r)
end;
echo "finisheD"
```

suricactus commented on docs/docs/examples.md:
````diff
@@ -0,0 +1,265 @@
+    qfieldcloud-cli delete-files "123e4567-e89b-12d3-a456-426614174000" "DCIM\tree-202411202334943.jpg"
+    ```
+
+### **Other Useful QFieldCloud CLI Commands**
````
This will die, we move all the commands up there.

SeqLaz commented:
> Ok, almost there. A few comments:
> 
>     * Rename "Linux" to "Bash" and "Windows" to "PowerShell"
> 
>     * Add both version for all commands.
> 
>     * Also put somewhere an advanced example using `qfieldcloud-cli --json login | jq .session_token` or something like this.

Hey @suricactus, I addressed all the suggestions and made some improvements to the redaction documentation.

suricactus commented:
Thanks!
